packages/go-read-write/read-write-example.go

```
@@ -0,0 +1,75 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+)
+
+func main() {
+	// READ DATA (QUERY)
+	// jsonData := map[string]string{
+	//     "query": `
+	//         { 
+	//             foo(id: 1) {
+	//                 id,
+	//                 bar
+	//             }
+	//         }
+	//     `,
+	// }
+
+
+	// WRITE DATA (MUTATION)
+	jsonData := map[string]interface{}{
+		"query": `
+			mutation ($foo: CreateFooInput) {
+				createFoo(foo: $foo) {
+					success
+					errors {
+						message
+						... on InvalidRecordError {
+							validationErrors {
+								apiIdentifier
+								message
+							}
+						}
+					}
+					foo {
+						id
+						bar
+					}
+				}
+			},
+		
+		`,
+		"variables": map[string]interface{}{
+			"foo": map[string]int{
+				"bar": 1,
+			},
+		},
+	}
+	jsonValue, _ := json.Marshal(jsonData)
+
+	// send the GraphQL request to your Gadget app
+	request, _ := http.NewRequest("POST", "https://<your-gadget-slug>/api/graphql", bytes.NewBuffer(jsonValue))
```

We're pretty inconsistent with error handling here, we ignore every error except the one from `Client.Do`, it would be good to properly handle errors in case they make a mistake with the JSON data or slug instead of silently swallowing it.

A decent error handler could look like:

```
request, err := http.NewRequest()
if err != nil {
    log.Fatal(err)
}
```

`log.Fatal` will end the process instead of continuing in the case of an error.